server/src/test/java/com/cloud/network/Ipv6ServiceImplTest.java

```diff
@@ -0,0 +1,781 @@
+            Ipv6GuestPrefixSubnetNetworkMapVO map = (Ipv6GuestPrefixSubnetNetworkMapVO)invocation.getArguments()[0];
+            persistedPrefixSubnetMap.add(map);
+            return map;
+        });
+        PowerMockito.mockStatic(ApiDBUtils.class);
+        Mockito.when(ApiDBUtils.findZoneById(Mockito.anyLong())).thenReturn(Mockito.mock(DataCenterVO.class));
+    }
+
+    private DataCenterGuestIpv6PrefixVO prepareMocksForIpv6Subnet() {
+        final long prefixId = 1L;
+        DataCenterGuestIpv6PrefixVO prefix = Mockito.mock(DataCenterGuestIpv6PrefixVO.class);
+        Mockito.when(prefix.getId()).thenReturn(prefixId);
+        Mockito.when(prefix.getPrefix()).thenReturn(ipv6Prefix);
+        List<Ipv6GuestPrefixSubnetNetworkMapVO> subnets = new ArrayList<>();
+        Ipv6GuestPrefixSubnetNetworkMapVO subnetMap = new Ipv6GuestPrefixSubnetNetworkMapVO(prefixId, "subnet", 1L, Ipv6GuestPrefixSubnetNetworkMap.State.Allocated);
+        subnets.add(subnetMap);
+        subnetMap = new Ipv6GuestPrefixSubnetNetworkMapVO(1L, "subnet", 2L, Ipv6GuestPrefixSubnetNetworkMap.State.Allocated);
+        subnets.add(subnetMap);
+        Mockito.when(ipv6GuestPrefixSubnetNetworkMapDao.listUsedByPrefix(prefixId)).thenReturn(subnets);
+        return prefix;
+    }
+
+    @Test
+    public void testGetUsedTotalIpv6SubnetForPrefix() {
+        DataCenterGuestIpv6PrefixVO prefix = prepareMocksForIpv6Subnet();
+        Pair<Integer, Integer> results = ipv6Service.getUsedTotalIpv6SubnetForPrefix(prefix);
+        Assert.assertEquals(2, results.first().intValue());
+        Assert.assertEquals(4, results.second().intValue());
+    }
+
+    @Test
+    public void testNoPrefixesGetUsedTotalIpv6SubnetForZone() {
+        final long zoneId = 1L;
+        final List<DataCenterGuestIpv6PrefixVO> prefixes = new ArrayList<>();
+        Mockito.when(dataCenterGuestIpv6PrefixDao.listByDataCenterId(zoneId)).thenReturn(prefixes);
+        Pair<Integer, Integer> results = ipv6Service.getUsedTotalIpv6SubnetForZone(zoneId);
+        Assert.assertEquals(0, results.first().intValue());
+        Assert.assertEquals(0, results.second().intValue());
+    }
+
+    @Test
+    public void testGetUsedTotalIpv6SubnetForZone() {
+        final long zoneId = 1L;
+        final List<DataCenterGuestIpv6PrefixVO> prefixes = new ArrayList<>();
+        DataCenterGuestIpv6PrefixVO prefix = prepareMocksForIpv6Subnet();
+        prefixes.add(prefix);
+        prefixes.add(prefix);
+        Mockito.when(dataCenterGuestIpv6PrefixDao.listByDataCenterId(zoneId)).thenReturn(prefixes);
+        Pair<Integer, Integer> results = ipv6Service.getUsedTotalIpv6SubnetForZone(zoneId);
+        Assert.assertEquals(4, results.first().intValue());
+        Assert.assertEquals(8, results.second().intValue());
+    }
+
+    @Test(expected = ResourceAllocationException.class)
+    @DB
```

Comment:
Unit tests using DB are a bit worrying. Do these tests reallly need this annotation?
and if so; Are these tests really needed/not covered by the integration tests?